bigquery/query.go

```diff
@@ -366,6 +366,27 @@ func (q *Query) newJob() (*bq.Job, error) {
 	}, nil
 }
 
+// ReadAsArrowObjects submits a query for execution and returns the results via an ArrowIterator.
+// As a prerequisite storage read client should be enabled.
+// It will always use jobs.insert path so may not be efficient for small queries.
+func (q *Query) ReadAsArrowObjects(ctx context.Context) (it *ArrowIterator, err error) {
+	if !q.client.isStorageReadAvailable() {
+		return nil, errors.New("bigquery: require storage read client for fetching records as arrow objects")
+	}
+
+	q.forceStorageAPI = true
+	rowIter, err := q.Read(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if rowIter.arrowIterator == nil {
+		return nil, errors.New("bigquery: results not available as arrow records")
+	}
```

Comment:
is there a particular kind of query / results that can't be returned as an arrow record batch stream? Can we document those?

Reply:
As I understood, this is not possible right now since we enforce the usage of `storageAPI`. I added this as a defensive check. I can remove this if this is adding confusion.

Reply:
@alvarowolfx As long as you have a schema, you can use https://pkg.go.dev/github.com/apache/arrow/go/v13/arrow/array#NewJSONReader or the individual `FromJSON` functions etc.

In addition, all of the `Builder`s have an `UnmarshalJSON` method which will append the unmarshalled json to the builder.